apps/wpcom-block-editor/src/utils.js

```diff
@@ -29,10 +29,33 @@ export function sendMessage( message ) {
 }
 
 /**
- * Indicates if the block editor has been initialized with blocks.
+ * Indicates if the block editor has been initialized.
  *
  * @returns {Promise} Promise that resolves when the editor has been initialized.
  */
+export const isEditorReady = async () =>
+	new Promise( ( resolve ) => {
+		const unsubscribe = subscribe( () => {
+			// Calypso sends the message as soon as the iframe is loaded, so we
+			// need to be sure that the editor is initialized and the core blocks
+			// registered. There is an unstable selector for that, so we use
+			// `isCleanNewPost` otherwise which is triggered when everything is
+			// initialized if the post is new.
+			const editorIsReady = select( 'core/editor' ).__unstableIsEditorReady
+				? select( 'core/editor' ).__unstableIsEditorReady()
+				: select( 'core/editor' ).isCleanNewPost();
```

Comment:
I think this is the culprit for the Site Editor not loading.  The unstable selector `__unstableIsEditorReady` doesn't seem to be updated by the site editor loading.

![Screen Shot 2022-01-07 at 11 24 33 AM](https://user-images.githubusercontent.com/28742426/148574232-63d26a21-2af5-4f9c-a183-2609bc1005a2.png)

Taken from a loaded site editor instance.